.github/workflows/agw-workflow.yml

```diff
@@ -0,0 +1,298 @@
+---
+name: agw-workflow
+
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches:
+      - master
+    paths:
+      - 'orc8r/**'
+      - 'lte/**'
+  pull_request:
+    branches:
+      - master
+jobs:
+  lte-test:
+    runs-on: ubuntu-latest
+    env:
+      MAGMA_ROOT: "${{ github.workspace }}"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+      - name: Install libraries and dependencies
+        env:
+          PYTHON_BUILD: "${{ env.MAGMA_ROOT }}/build"
+          PIP_CACHE_HOME: "${{ env.MAGMA_ROOT }}/.pipcache"
+          MAGMA_DEV_MODE: 1
+          SKIP_SUDO_TESTS: 1
+          CODEGEN_ROOT: "${{ env.MAGMA_ROOT }}/.codegen"
+          SWAGGER_CODEGEN_JAR: "${{ env.MAGMA_ROOT }}/.codegen/swagger-codegen-cli.jar"
+        run: |
+            mkdir -p /var/tmp/test_results
+            mkdir -p /var/tmp/codecovs
+            sudo apt-get update -y
+            sudo apt-get install -y libsystemd-dev pkg-config curl zip unzip
+            sudo apt-get install -y virtualenv python-babel python-dev build-essential python3-setuptools python-setuptools autogen autoconf libtool python3-apt python3-requests python3-pip python-protobuf
+            pip3 install setuptools==49.6.0
+            sudo curl -Lfs https://github.com/google/protobuf/releases/download/v3.1.0/protoc-3.1.0-linux-x86_64.zip -o protoc3.zip
+            sudo unzip protoc3.zip -d protoc3
+            sudo mv protoc3/bin/protoc /bin/protoc
+            sudo chmod a+rx /bin/protoc
+            sudo mv protoc3/include/google /usr/include/
+            sudo chmod -R a+Xr /usr/include/google
+            sudo rm -rf protoc3.zip protoc3
+            sudo -E mkdir ${{ env.CODEGEN_ROOT }}
+            sudo -E wget https://repo1.maven.org/maven2/io/swagger/swagger-codegen-cli/2.2.3/swagger-codegen-cli-2.2.3.jar -O ${{ env.SWAGGER_CODEGEN_JAR }}
+            # MAGMA_ROOT needs to be exported so that python.py is able to recognize it
+            export MAGMA_ROOT=${{ env.MAGMA_ROOT }}
+            sudo -E chmod -R 755 ${{ env.CODEGEN_ROOT }}
+            echo "$MAGMA_ROOT"
+            echo "Run test_all"
+            sudo -E make -C ${{ env.MAGMA_ROOT }}/lte/gateway/python test_all
```

Comment:
Can you separate a bit this huge step into multiple.
ie, one for the packages, one for protoc3, one for swagger codegen and one for the test run

Reply:
sure!